scripts/histmakers/mz_dilepton.py

```diff
@@ -0,0 +1,229 @@
+else:
+    logging.info("Using smoothed scale factors and uncertainties")
+    muon_efficiency_helper, muon_efficiency_helper_syst, muon_efficiency_helper_stat = wremnants.make_muon_efficiency_helpers_smooth(filename = args.sfFile,
+                                                                                                                                     era = era,
+                                                                                                                                     max_pt = args.pt[2],
+                                                                                                                                     is_w_like = True)
+logging.info(f"SF file: {args.sfFile}")
+
+pileup_helper = wremnants.make_pileup_helper(era = era)
+
+mc_jpsi_crctn_helper, data_jpsi_crctn_helper = muon_validation.make_jpsi_crctn_helpers(args.muonCorr)
+
+mc_calibration_helper, data_calibration_helper, calibration_uncertainty_helper = muon_calibration.make_muon_calibration_helpers()
+
+smearing_helper = muon_calibration.make_muon_smearing_helpers() if args.smearing else None
+
+bias_helper = muon_calibration.make_muon_bias_helpers(args.muonCorr, args.smearing) if args.bias_calibration else None
+
+corr_helpers = theory_corrections.load_corr_helpers(common.vprocs, args.theory_corr)
+
+# recoil initialization
+if not args.no_recoil:
+    from wremnants import recoil_tools
+    recoilHelper = recoil_tools.Recoil("highPU", flavor="mumu", met=args.met)
+
+
+def build_graph(df, dataset):
+    logging.info(f"build graph for dataset: {dataset.name}")
+    results = []
+    isW = dataset.name in common.wprocs
+    isZ = dataset.name in common.zprocs
+
+    if dataset.is_data:
+        df = df.DefinePerSample("weight", "1.0")
+    else:
+        df = df.Define("weight", "std::copysign(1.0, genWeight)")
+
+    weightsum = df.SumAndCount("weight")
+
+    df = df.Filter("HLT_IsoTkMu24 || HLT_IsoMu24")
+
+    df = muon_selections.veto_electrons(df)
+    df = muon_selections.apply_met_filters(df)
+
+    cvh_helper = data_calibration_helper if dataset.is_data else mc_calibration_helper
+    jpsi_helper = data_jpsi_crctn_helper if dataset.is_data else mc_jpsi_crctn_helper
+
+    df = muon_calibration.define_corrected_muons(df, cvh_helper, jpsi_helper, args.muonCorr, dataset, smearing_helper, bias_helper)
+
+    df = muon_selections.select_veto_muons(df, nMuons=2)
+    df = muon_selections.select_good_muons(df, nMuons=2, use_trackerMuons=args.trackerMuons, use_isolation=True)
+
+    df = muon_selections.define_trigger_muons(df)
+
+    df = muon_selections.select_standalone_muons(df, dataset, args.trackerMuons, "trigMuons")
+    df = muon_selections.select_standalone_muons(df, dataset, args.trackerMuons, "nonTrigMuons")
+
+    df = muon_selections.select_triggermatched_muon(df, dataset, "trigMuons_eta0", "trigMuons_phi0")
+
+    df = df.Filter("trigMuons_pt0 > 26.") # should there also be an upper cut?
```

Comment:
Would be good to have these lines factored into a separate function to avoid the risk of the two going out of sync if the dilepton and single lepton histmakers are split off

Reply:
Do we need to keep the pt on the trigger muon in the underflow and overflow for the wlike fit? Otherwise I would just always cut those away

Reply:
For the main lepton, which is the triggering one and defines the acceptance of the histogram, we don't explicitly cut because in that way we can implement variations acting as pt shifts without creating undesired features at the edge of the histogram pt range. So we let the histogram range implicitly act as a pt cut (and we never use the bin contents for overflow bins for pt anywhere at the moment)

in #123 I added the lower and upper pt cut on the second muon (non triggering), and currently they are set automatically as the same pt range as the first lepton. This might need to be more general in case we don't use asymmetric ranges.

Reply:
see here:
https://github.com/WMass/WRemnants/blob/main/wremnants/muon_selections.py#L54

I only updated this one for the mz_wlike though